pkg/directory/farms_handlers.go

```diff
@@ -152,6 +153,50 @@ func (f *FarmAPI) deleteNodeFromFarm(r *http.Request) (interface{}, mw.Response)
 	return nil, mw.Ok()
 }
 
+func (f *FarmAPI) setNodeDedicated(r *http.Request) (interface{}, mw.Response) {
+	farmID := getFarmID(r.Context())
+
+	data := struct {
+		nodeID string
```

Comment:
`nodeID` is private, hence it can't be "deseralized" this must be named `NodeID string json:"node_id"`

Reply:
Same for he UserID